### PR TITLE
Remote fetch array node

### PR DESCRIPTION
**Pull request by wild-endeavor**
## Tracking issue
https://github.com/flyteorg/flyte/issues/5359

## What changes were proposed in this pull request?
The fundamental issue leading to the error was in the order of the args in FlyteArrayNode promote_from_model.  That's been fixed and support for array nodes added.

* The `NodeExecutionMetaData` model hadn't been updated in a while (lacking some fields) but it's not widely used so I just removed it entirely opting for the raw proto generated file.
* Rename an import

## How was this patch tested?
* Added a sandbox test that doesn't run for when we have a nice integration testing framework.

## Check all the applicable boxes 

- [ ] I updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] All commits are signed-off.

### CODE REVIEW

File: flytekit/models/node_execution.py [hunk 1]
```diff
@@ -296,22 +262,23 @@ def closure(self):
...
 
     @classmethod
-    def from_flyte_idl(cls, p: _node_execution_pb2.NodeExecution) -> "NodeExecution":
+    def from_flyte_idl(cls, p: admin_node_execution_pb2.NodeExecution) -> "NodeExecution":
         return cls(
             id=_identifier.NodeExecutionIdentifier.from_flyte_idl(p.id),
             input_uri=p.input_uri,
             closure=NodeExecutionClosure.from_flyte_idl(p.closure),
-            metadata=NodeExecutionMetaData.from_flyte_idl(p.metadata),
+            # metadata=NodeExecutionMetaData.from_flyte_idl(p.metadata),
```
Comment (cosmicBboy) [inline flytekit/models/node_execution.py:282]: remove?

File: flytekit/remote/entities.py [hunk 2]
```diff
@@ -472,6 +477,7 @@ def promote_from_model(
             flyte_gate_node = FlyteGateNode.promote_from_model(model.gate_node)
         elif model.array_node is not None:
             flyte_array_node = FlyteArrayNode.promote_from_model(model.array_node)
+            print(flyte_array_node)
```
Comment (cosmicBboy) [inline flytekit/remote/entities.py:480]: remove print

File: flytekit/remote/remote.py [hunk 3]
```diff
@@ -2101,7 +2101,7 @@ def sync_node_execution(
                 sub_node_mapping = {n.id: n for n in sub_flyte_workflow.flyte_nodes}
                 execution._underlying_node_executions = [
                     self.sync_node_execution(FlyteNodeExecution.promote_from_model(cne), sub_node_mapping)
-                    for cne in child_node_executions
+                    for cne in child_node_executiFsync_execons
```
Comment (cosmicBboy) [inline flytekit/remote/remote.py:2104]: where is `child_node_executiFsync_execons` coming from? also kind of a strange-looking variable name...

Comment (wild-endeavor) [reply flytekit/remote/remote.py:2104]: uh, not sure how that happened, thanks

**Review by eapolinario — APPROVED** [review]

**Pull request merged**